views/reports/monthly_report.php

```diff
@@ -1,1248 +1,1985 @@
+		$pNewRes = pg_fetch_result($db->query($pNewQuery ,[]), 0, 0);
+		$pDupRes = $pMonthRes - $pNewRes;
+		$pYearQuery = $pBaseQuery . "WHERE " . $yearWhere;
+		$pYearRes = pg_fetch_result($db->query($pYearQuery ,[]), 0, 0);
+		
+		#Gender Where clauses
+		$femaleWhere = "(sex = 'Female')";
+		$maleWhere = "(sex = 'Male')";
+		
+		#Male Queries
+		$query = $pMonthQuery . " AND $maleWhere ";
+		$mMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $maleWhere ";
+		$mNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$mDupRes = $mMonthRes - $mNewRes;
+		$query = $pYearQuery . " AND $maleWhere ";
+		$mYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#Female Queries
+		$query = $pMonthQuery . " AND $femaleWhere ";
+		$fMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $femaleWhere ";
+		$fNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$fDupRes = $fMonthRes - $fNewRes;
+		$query = $pYearQuery . " AND $femaleWhere ";
+		$fYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#Age Where clauses
+		$_20Where = "(date_part('year', AGE(dateofbirth)) >= 20)
+				AND (date_part('year', AGE(dateofbirth)) <= 40)";
+		$_41Where = "(date_part('year', AGE(dateofbirth)) >= 41)
+				AND (date_part('year', AGE(dateofbirth)) <= 64)";
+		$_65Where = "(date_part('year', AGE(dateofbirth)) >= 65)";
+		
+		#20-41 Queries
+		$query = $pMonthQuery . " AND $_20Where ";
+		$_20MonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $_20Where ";
+		$_20NewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$_20DupRes = $_20MonthRes - $_20NewRes;
+		$query = $pYearQuery . " AND $_20Where ";
+		$_20YearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#41-64 Queries
+		$query = $pMonthQuery . " AND $_41Where ";
+		$_41MonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $_41Where ";
+		$_41NewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$_41DupRes = $_41MonthRes - $_41NewRes;
+		$query = $pYearQuery . " AND $_41Where ";
+		$_41YearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#65+ Queries
+		$query = $pMonthQuery . " AND $_65Where ";
+		$_65MonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $_65Where ";
+		$_65NewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$_65DupRes = $_65MonthRes - $_65NewRes;
+		$query = $pYearQuery . " AND $_65Where ";
+		$_65YearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#Ethnicity Where clauses
+		$afrAmWhere = " race = 'African American' ";
+		$natAmWhere = " race = 'Native American' ";
+		$pacIslWhere = " race = 'Pacific Islander' ";
+		$caucWhere = " race = 'Caucasian' ";
+		$multRacWhere = " race = 'Multi Racial' ";
+		$latWhere = " race = 'Latino' ";
+		$otherRacWhere = " race = 'Other' ";
+		
+		#Caucasian Queries
+		$query = $pMonthQuery . " AND $caucWhere ";
```

Comment:
I would suggest making a generic function for the race queries and maybe even other queries that you are duplicating. For example, you can use the following generic function for the race queries:
```php
$raceQuery = function($where, &$monthRes, &$newRes, &$dupRes, &$yearRes)
        use ($db, $pMonthQuery, $pNewQuery, $pYearQuery) {
    $query = $pMonthQuery . " AND $where ";
    $monthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
    $query = $pNewQuery . " AND $where ";
    $newRes = pg_fetch_result($db->query($query ,[]), 0, 0);
    $dupRes = $monthRes - $newRes;
    $query = $pYearQuery . " AND $where ";
    $yearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
};

#Caucasian Queries
$raceQuery($caucWhere, $caucMonthRes, $caucNewRes, $caucDupRes, $caucYearRes);

#African American Queries
$raceQuery($afrAmWhere, $afAmMonthRes, $afAmNewRes, $afAmDupRes, $afAmYearRes);

...
```

The "&" in the function definition is pass-by-reference so it'll store the result in the variable you pass in.

Reply:
Thank you, this is exactly what I was looking for. Didn't know how to do it in PHP. I'll implement it now.